scheduler/core/programprovider/ocs/ocsprogramprovider.py

```diff
@@ -763,109 +773,111 @@ def autocorr_lag(x):
         n_offsets = 0
         n_pattern = offset_lag
         prev = -1
+        step_use = -1
         for step_id, step in enumerate(sequence):
             next_atom = False
-
             observe_class = step[OcsProgramProvider._AtomKeys.OBS_CLASS]
-            step_time = step[OcsProgramProvider._AtomKeys.TOTAL_TIME] / 1000
-
-            # Any wavelength/filter change is a new atom
-            if step_id == 0 or (step_id > 0 and wavelengths[step_id] != wavelengths[step_id - 1]):
-                next_atom = True
-                # logger.info('Atom for wavelength change')
-                # print(f'\t\t\t Atom for wavelength change')
-
-            # A change in exposure time or coadds is a new atom for science exposures
-            # print(f'\t\t\t {step[OcsProgramProvider._AtomKeys.OBSERVE_TYPE].upper()}')
-            if step[OcsProgramProvider._AtomKeys.OBSERVE_TYPE].upper() not in OcsProgramProvider._OBSERVE_TYPES:
-                if (prev >= 0 and observe_class.upper() == ObservationClass.SCIENCE.name and step_id > 0 and
-                        (exposure_times[step_id] != exposure_times[prev] or coadds[step_id] != coadds[prev])):
+            if observe_class.upper() not in [ObservationClass.ACQ.name, ObservationClass.ACQCAL.name]:
```

Comment:
Here's another place where having a function to check this would avoid repeated code.